Make it possible to hard-code touch point radius

hortont424 opened the pull request:
#### 7e5ac4506822fab82a65230a1cc9a679369fe99f
<pre>
Make it possible to hard-code touch point radius
<a href="https://bugs.webkit.org/show_bug.cgi?id=257147">https://bugs.webkit.org/show_bug.cgi?id=257147</a>
rdar://109067957

Reviewed by Megan Gardner, Wenson Hsieh and Aditya Keerthi.

* Source/WebKit/Shared/ios/NativeWebTouchEventIOS.mm:
(WebKit::convertTouchType):
(WebKit::radiusForTouchPoint):
(WebKit::NativeWebTouchEvent::extractWebTouchPoint):
Add a build flag that allows fixing the touch point radius.
Drive-by remove some staging.

Canonical link: <a href="https://commits.webkit.org/264383@main">https://commits.webkit.org/264383@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e873394762084a99529bd83bf813920783f70afb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/7453 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7723 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7895 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/9089 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/7656 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/9655 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7640 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/10546 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7584 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/8701 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/6905 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/9200 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/6008 "Passed tests") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/14510 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/7235 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/6896 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/10208 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7391 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/6048 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6737 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1767 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10948 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7124 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/Shared/ios/NativeWebTouchEventIOS.mm:
```diff
@@ -83,13 +82,21 @@ static inline WebEventType webEventTypeForUIWebTouchEventType(UIWebTouchEventTyp
+static CGFloat radiusForTouchPoint(const _UIWebTouchPoint& touchPoint)
+{
+#if ENABLE(FIXED_IOS_TOUCH_POINT_RADIUS)
+    return 12.1;
```
Do we need an `UNUSED_PARAM` in here?

hortont424 replied:
Not in WK, right? Only WebCore

megangardner commented on Source/WebKit/Shared/ios/NativeWebTouchEventIOS.mm:
```diff
@@ -70,7 +70,6 @@ static inline WebEventType webEventTypeForUIWebTouchEventType(UIWebTouchEventTyp
     }
 }
 
-#if defined UI_WEB_TOUCH_EVENT_HAS_STYLUS_DATA && UI_WEB_TOUCH_EVENT_HAS_STYLUS_DATA
```
I don't see why in your code this needs to be exposed, so I'm assuming it's for a related/follow up reason?

hortont424 replied:
"Drive by" removing long-unneeded guards that nobody ever cleaned up.

hortont424 committed:
Make it possible to hard-code touch point radius
https://bugs.webkit.org/show_bug.cgi?id=257147
rdar://109067957

Reviewed by Megan Gardner, Wenson Hsieh and Aditya Keerthi.

* Source/WebKit/Shared/ios/NativeWebTouchEventIOS.mm:
(WebKit::convertTouchType):
(WebKit::radiusForTouchPoint):
(WebKit::NativeWebTouchEvent::extractWebTouchPoint):
Add a build flag that allows fixing the touch point radius.
Drive-by remove some staging.

Canonical link: https://commits.webkit.org/264383@main
